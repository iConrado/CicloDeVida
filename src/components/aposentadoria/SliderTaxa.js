import React from 'react';

import { 
  View, 
  Text } from 'react-native';
import Slider from 'react-native-slider';

import styles from '../functions/styles';

export default class SliderTaxa extends React.Component {
  constructor(props) {
    super(props);
    this.state = { tmpRentab: props.inicial ? props.inicial : 0 };
  }

  render() {
    return (
      <View style={styles.viewVertical}>
        <View style={styles.viewHorizontal}>
          <Text style={styles.label}>Taxa de rentabilidade estimada (a.a.):</Text>
        </View>
        <View style={styles.viewHorizontal}>
          <Slider
            style={styles.aposent_slider}
            minimumValue={0}
            maximumValue={20}
            step={0.1}
            minimumTrackTintColor='#14567A'
            thumbTintColor='#14567A'
            value={this.state.tmpRentab}
            onValueChange={(value) => this.setState({ tmpRentab: value })}
            onSlidingComplete={(value) => this.props.retorno(value)}
          />
          <View style={styles.aposent_viewCentral}>
            <Text style={styles.aposent_txDireita}>
              {this.state.tmpRentab.toFixed(1).replace('.', ',')}%
            </Text>
          </View>
        </View>
      </View>
    );
  }
}
