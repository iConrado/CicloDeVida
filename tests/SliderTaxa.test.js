import React from 'react';
import { shallow } from 'enzyme';
import SliderTaxa from '../src/components/aposentadoria/SliderTaxa';
import Slider from 'react-native-slider';

describe('SliderTaxa', () => {
  let ret;
  const retorno = valor => { ret = valor; };
  const app = shallow(<SliderTaxa inicial={50} retorno={retorno} />);
  const app2 = shallow(<SliderTaxa retorno={retorno} />);

  it('renders correctly', () => {  
    expect(app).toMatchSnapshot();
  });

  it('slider sem o valor inicial', () => {  
    app2.find(Slider).at(0).simulate('valueChange', 30);
    expect(app2.state('tmpRentab')).toBe(30);
  });

  it('zera o slider', () => {  
    app.find(Slider).at(0).simulate('valueChange', 0);
    expect(app.state('tmpRentab')).toBe(0);
  });

  it('altera o valor do slider', () => {  
    app.find(Slider).at(0).simulate('valueChange', 70);
    expect(app.state('tmpRentab')).toBe(70);
  });

  it('retorna valor do slider', () => {  
    app.find(Slider).at(0).simulate('slidingComplete', 70);
    expect(ret).toBe(70);
  });
});
