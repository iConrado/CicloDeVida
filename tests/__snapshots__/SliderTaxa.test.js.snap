// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderTaxa renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderTaxa
    inicial={50}
    retorno={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Taxa de rentabilidade estimada (a.a.):
          </Text>
        </View>,
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Slider
            animationType="timing"
            debugTouchArea={false}
            maximumTrackTintColor="#b3b3b3"
            maximumValue={20}
            minimumTrackTintColor="#14567A"
            minimumValue={0}
            onSlidingComplete={[Function]}
            onValueChange={[Function]}
            step={0.1}
            style={
              Object {
                "height": 40,
                "width": "70%",
              }
            }
            thumbTintColor="#14567A"
            thumbTouchSize={
              Object {
                "height": 40,
                "width": 40,
              }
            }
            value={50}
          />
          <View
            style={
              Object {
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              50,0
              %
            </Text>
          </View>
        </View>,
      ],
      "style": Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Taxa de rentabilidade estimada (a.a.):
          </Text>,
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Taxa de rentabilidade estimada (a.a.):",
            "style": Object {
              "color": "#666",
            },
          },
          "ref": null,
          "rendered": "Taxa de rentabilidade estimada (a.a.):",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={20}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={0.1}
              style={
                Object {
                  "height": 40,
                  "width": "70%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />,
            <View
              style={
                Object {
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                50,0
                %
              </Text>
            </View>,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationType": "timing",
              "debugTouchArea": false,
              "maximumTrackTintColor": "#b3b3b3",
              "maximumValue": 20,
              "minimumTrackTintColor": "#14567A",
              "minimumValue": 0,
              "onSlidingComplete": [Function],
              "onValueChange": [Function],
              "step": 0.1,
              "style": Object {
                "height": 40,
                "width": "70%",
              },
              "thumbTintColor": "#14567A",
              "thumbTouchSize": Object {
                "height": 40,
                "width": 40,
              },
              "value": 50,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                50,0
                %
              </Text>,
              "style": Object {
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "50,0",
                  "%",
                ],
                "style": Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              },
              "ref": null,
              "rendered": Array [
                "50,0",
                "%",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Taxa de rentabilidade estimada (a.a.):
            </Text>
          </View>,
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={20}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={0.1}
              style={
                Object {
                  "height": 40,
                  "width": "70%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />
            <View
              style={
                Object {
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                50,0
                %
              </Text>
            </View>
          </View>,
        ],
        "style": Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Taxa de rentabilidade estimada (a.a.):
            </Text>,
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Taxa de rentabilidade estimada (a.a.):",
              "style": Object {
                "color": "#666",
              },
            },
            "ref": null,
            "rendered": "Taxa de rentabilidade estimada (a.a.):",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={20}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={0.1}
                style={
                  Object {
                    "height": 40,
                    "width": "70%",
                  }
                }
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />,
              <View
                style={
                  Object {
                    "height": 40,
                    "justifyContent": "center",
                    "width": "30%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  50,0
                  %
                </Text>
              </View>,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animationType": "timing",
                "debugTouchArea": false,
                "maximumTrackTintColor": "#b3b3b3",
                "maximumValue": 20,
                "minimumTrackTintColor": "#14567A",
                "minimumValue": 0,
                "onSlidingComplete": [Function],
                "onValueChange": [Function],
                "step": 0.1,
                "style": Object {
                  "height": 40,
                  "width": "70%",
                },
                "thumbTintColor": "#14567A",
                "thumbTouchSize": Object {
                  "height": 40,
                  "width": 40,
                },
                "value": 50,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  50,0
                  %
                </Text>,
                "style": Object {
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "50,0",
                    "%",
                  ],
                  "style": Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "50,0",
                  "%",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
